test/unit/test_registry_get_request_timeout.py

```diff
@@ -0,0 +1,9 @@
+import unittest
+from requests.exceptions import Timeout
+
+from dbt.clients.registry import _get
+
+class testRegistryGetRequestTimeout(unittest.TestCase):
+    def test_registry_request_timeout(self):
+        # using non routable IP to test timeout logic in the _get function
+        self.assertRaises(Timeout, _get, '', 'http://10.255.255.1')
```

Comment:
Related to above, it looks like [the failing Windows test](https://dev.azure.com/fishtown-analytics/dbt/_build/results?buildId=3252&view=logs&j=1c778e8e-596b-5a0a-22c2-460364725d7c&t=1607d490-6a74-5e3f-97a9-72eb94e82d47) is skipping the `Timeout` error, retrying 5 times, and finally raising `dbt.exceptions.RegistryException: Unable to connect to registry hub`.

Let's do one of:
- Figure out why Windows is skipping the `Timeout` exception and jumping into retries
- Amend the code above to raise a consistent retry-based exception
- Edit this test to check for either `Timeout` or `RegistryException`

Reply:
Based on the above update to the timeout error handling I went ahead and made a generic error catching test.